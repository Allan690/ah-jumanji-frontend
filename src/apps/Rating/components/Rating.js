/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import StarRatings from 'react-star-ratings';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {fetchArticleRating, postArticleRating} from '../actions/ratingActions';


class Rating extends Component {
  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  componentDidMount() {
    const { fetchArticleRating } = this.props;
    fetchArticleRating();
  }

  handleClick(newRating){
    const ratingData = {
      "rating": newRating
    };
    const { postArticleRating } = this.props;
    postArticleRating(ratingData);
  }
  render() {
    const {Rating} = this.props;
    const rated = Rating.rating == "Article has no ratings" ? 0 : Rating.rating;
    return (
      <StarRatings
        rating={rated}
        starDimension="20px"
        starSpacing="2px"
        starRatedColor="orange"
        starEmptyColor="lightgray"
        starHoverColor="orange"
        changeRating={this.handleClick}
      />
    );
  }
}

Rating.propTypes = {
  fetchArticleRating: PropTypes.func.isRequired,
  Rating: PropTypes.object.isRequired,
  postArticleRating: PropTypes.func.isRequired,

};

function mapStateToProps(state) {
  return {
    Rating: state.ratingReducer.averageRating
  };
}
export default connect(mapStateToProps, { fetchArticleRating, postArticleRating })(Rating);