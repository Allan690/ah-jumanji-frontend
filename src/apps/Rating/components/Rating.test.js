import React from 'react';

import { shallow } from 'enzyme';
import Rating from './Rating';



describe('Rating page', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Rating debug />);

    expect(component).toMatchSnapshot();
  });

});
