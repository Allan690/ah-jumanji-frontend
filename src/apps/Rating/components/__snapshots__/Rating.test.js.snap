// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating page should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
